rules/migrate-from-system-web-to-modern-alternatives/rule.md

```diff
@@ -164,6 +176,22 @@ namespace Microsoft.AspNetCore.Http
 Figure: A .NET Framework specific implementation of the IHttpContextAccessor interface, only available when the target framework is .NET 4.7.2 or greater.
 :::
 
+::: greybox
+NOTE: Make sure your .csproj is correctly configured. Don't use 
+
+```xml
+<!-- .NET Framework reference for HttpContext -->
+<ItemGroup Condition="'$(TargetFramework)' == 'net472'">
+    <Reference Include="System.Web" />
+</ItemGroup>
+
+<!-- .NET reference for IHttpContextAccessor, already included if the project is WebApp -->
+<ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+</ItemGroup>
+```
+:::
+
```

Comment:
I recommend rewriting this greybox in higher-level concepts and explicitly stating why this is important. If I understand your intention correctly, you are stating that the project cannot include references to the `System.Web` and `Microsoft.AspNetCore.App` assemblies. Is the reason just to prevent a namespace collision?

Reply:
No, `System.Web` won't compile on .NET Standard and .NET Core 3.1+ (`Reference`) and `Microsoft.AspNetCore.App` won't compile on .NET Framework or .NET Standard (`FrameworkReference`). They are both framework-specific references unlike `PackageReference`.

Reply:
Resolving for now - see comment below